src/util/pdfToImages.ts

```diff
@@ -0,0 +1,25 @@
+import { createCanvas } from 'canvas'
+import { getDocument } from 'pdfjs-dist'
+
+/**
+ * Renders PDF pages as images.
+ */
+export default async function* pdfToImages(
+  pdfBytes: Buffer,
+  { scale = 2 } = {}
+): AsyncGenerator<ImageData> {
+  const canvas = createCanvas(0, 0)
+  const context = canvas.getContext('2d')
+  const pdf = await getDocument(pdfBytes).promise
+
+  for (let i = 1; i <= pdf.numPages; i++) {
+    const page = await pdf.getPage(i)
```

Comment:
I trust you've thought through this 1-indexing? Not sure what `pdf.getPage(0)` does.

Reply:
https://github.com/mozilla/pdf.js/blob/6ffcedc24bba417694a9d0e15eaf16cadf4dad15/src/display/api.js#L2457-L2463